client/src/main/java/org/asynchttpclient/netty/channel/ChannelManager.java

```diff
@@ -488,4 +493,28 @@ public ChannelPool getChannelPool() {
     public EventLoopGroup getEventLoopGroup() {
         return eventLoopGroup;
     }
+
+    public ClientStats getClientStats() {
+        final Map<String, Long> totalConnectionsPerHost = openChannels
+                    .stream()
+                    .map(Channel::remoteAddress)
+                    .filter(a -> a.getClass() == InetSocketAddress.class)
+                    .map(a -> (InetSocketAddress) a)
```

Comment:
Yeah, the filter beforehand means we're not going to see an exception. I was just worried about leaving out other kinds of sockets from the stats, but it sounds like we're good.